services/terraformer/server/domain/usecases/build_infrastructure.go

```diff
@@ -14,66 +11,108 @@ import (
-	// clusters slice contains those Kubernetes and Loadbalancer clusters
-	// for which infrastructure will be built.
-	// The "clusters" slice is initialized with the desired Kubernetes cluster we want to build.
-	clusters := []Cluster{
-		kubernetes.K8Scluster{
-			ProjectName: request.ProjectName,
-
-			DesiredState: request.Desired,
-			CurrentState: request.Current,
-
-			AttachedLBClusters: request.DesiredLbs,
-		},
+	k8sCluster := &kubernetes.K8Scluster{
+		ProjectName:        request.ProjectName,
+		DesiredState:       request.Desired,
+		CurrentState:       request.Current,
+		AttachedLBClusters: request.DesiredLbs,
 	}
 
-	// LB clusters which appear in both request.CurrentLbClusters and request.DesiredLBClusters
-	// are appended to the "clusters" slice
+	var lbClusters []*loadbalancer.LBcluster
 	for _, desiredLBCluster := range request.DesiredLbs {
-		var commonLb *pb.LBcluster
+		var current *pb.LBcluster
 
 		for _, currentLbCluster := range request.CurrentLbs {
 			if desiredLBCluster.ClusterInfo.Name == currentLbCluster.ClusterInfo.Name {
-				commonLb = currentLbCluster
+				current = currentLbCluster
 				break
 			}
 		}
 
-		clusters = append(clusters,
-			loadbalancer.LBcluster{
-				DesiredState: desiredLBCluster,
-				CurrentState: commonLb,
+		lbClusters = append(lbClusters, &loadbalancer.LBcluster{
+			DesiredState: desiredLBCluster,
+			CurrentState: current,
+			ProjectName:  request.ProjectName,
+		})
+	}
 
-				ProjectName: request.ProjectName,
-			},
-		)
+	clusters := []Cluster{k8sCluster}
+	for _, lb := range lbClusters {
+		clusters = append(clusters, lb)
 	}
 
-	// Concurrently build infrastructure for each cluster in the "clusters" slice
-	err := utils.ConcurrentExec(clusters, func(cluster Cluster) error {
+	failed := make([]error, len(clusters))
+	err := utils.ConcurrentExec(clusters, func(idx int, cluster Cluster) error {
 		logger := utils.CreateLoggerWithProjectAndClusterName(request.ProjectName, cluster.Id())
 		logger.Info().Msg("Creating infrastructure")
 
 		if err := cluster.Build(logger); err != nil {
-			return fmt.Errorf("error while building the cluster %v : %w", cluster.Id(), err)
+			if errors.Is(err, loadbalancer.ErrCreateDNSRecord) {
+				cluster.UpdateCurrentState()
+			}
+			logger.Err(err).Msgf("Error encountered while building cluster")
+			failed[idx] = err
+			return err
 		}
 
+		cluster.UpdateCurrentState()
 		logger.Info().Msgf("Infrastructure successfully created for cluster")
 		return nil
 	})
+
 	if err != nil {
-		log.Err(err).Str("project", request.ProjectName).Msgf("Error encountered while building cluster")
-		return nil, fmt.Errorf("error while building cluster %s for project %s : %w", request.Desired.ClusterInfo.Name, request.ProjectName, err)
-	}
+		// Given a failure in the building process construct the response such that those that successfully
+		// build will have their current state = desired state.
```

Comment:
Will this also resolve #861?

Reply:
No this just handles the case where the deletion of the cluster would fail if only part of the infra was build.